### PR TITLE
Fix incorrect database relationship cardinality

### LINKED ISSUE #1562: Fix database relationship cardinality
ManyToMany relationships are used in the server for no reason.  This is a simple conversion.  Good time to update these due to Postgres migration.

### PR DESCRIPTION
**Pull request by kholdaway**
Closes #1562

### CODE REVIEW

File: quipucords/api/inspectresult/model.py [hunk 1]
```diff
@@ -19,21 +19,39 @@
...
+        return '{ id:%s, task_results:%s }' % (self.id, self.task_results)
+
+    class Meta:
+        """Metadata for model."""
+
+        verbose_name_plural = _(messages.PLURAL_JOB_INSPECT_RESULTS_MSG)
+
+
+class TaskInspectionResult(models.Model):
+    """The captured connection results from a scan."""
```
Comment (abaiken) [inline quipucords/api/inspectresult/model.py]: and this one.

File: quipucords/api/scan/serializer.py [hunk 2]
```diff
@@ -186,190 +186,172 @@ def create(self, validated_data):
...
+                        **extended_search)
+                extended_search_instance.save()
                 instance.options.enabled_extended_product_search = \
-                    extended_search
+                    extended_search_instance
                 instance.options.save()
+            else:
+                # Update existing instance
+                # Grab the new extended search options
+
```
Comment (abaiken) [inline quipucords/api/scan/serializer.py]: remove blank line

File: quipucords/api/inspectresult/model.py [hunk 3]
```diff
@@ -19,21 +19,39 @@
 from api.source.model import Source
 
 
-class RawFact(models.Model):
-    """A model of a raw fact."""
+class JobInspectionResult(models.Model):
+    """The results of a connection scan."""
```
[the remark this answers is not in the document]
Comment (kholdaway) [reply quipucords/api/inspectresult/model.py]: Lol.  That's been there a while.  Fixed.

File: quipucords/api/inspectresult/model.py (hunk 1 above)
Comment (kholdaway) [reply quipucords/api/inspectresult/model.py]: same

File: quipucords/api/scan/serializer.py (hunk 2 above)
Comment (kholdaway) [reply quipucords/api/scan/serializer.py]: Fixed.

File: quipucords/api/common/util.py [hunk 4]
```diff
@@ -65,6 +73,8 @@ def convert_to_boolean(value):
     :param value: The value to convert
     :return The value as a bool
     """
+    if isinstance(value, bool):
```
Comment (abaiken) [inline quipucords/api/common/util.py]: I don't think we need to do this check since it is inside of function is_boolean which we immediately call

Comment (kholdaway) [reply quipucords/api/common/util.py]: we do or it will explode because bool has not method `value.lower() `

**Pull request merged**